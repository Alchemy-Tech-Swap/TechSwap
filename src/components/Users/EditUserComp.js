import React from 'react';
import './Users.css';

export default function EditUserComp({ label, name, onChange, slack_id, linkedin_url, onSubmit }) {
  return (
    <div>
      <div>EDIT</div>
      <div>{label || 'User Info'}</div>
      <form onSubmit={(e) => onSubmit(e)}>
        <label htmlFor="name">Name: </label>
        <input id="name" name="name" type="text" value={name} onChange={(e) => onChange(e)} />
        <br />
        <label htmlFor="slack_id">Slack: </label>
        <input
          id="slack_id"
          name="slack_id"
          type="text"
          value={slack_id}
          onChange={(e) => onChange(e)}
        />
        <br />
        <label htmlFor="linkedin_url">LinkedIn: </label>
        <input
          id="linkedin_url"
          name="linkedin_url"
          type="text"
          value={linkedin_url}
          onChange={(e) => onChange(e)}
        />
        <button>Submit</button>
      </form>
    </div>
  );
}
