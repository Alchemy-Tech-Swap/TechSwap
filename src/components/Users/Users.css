body {
  color: white;
  font-weight: bold;
  margin-top: 10%;
}
