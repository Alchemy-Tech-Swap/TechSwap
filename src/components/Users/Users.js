import React, { useState, useEffect } from 'react';

export default function Users({ user }) {
  return (
    <div>
      <h1>Profile</h1>
      <div>Name: {user.name}</div>
      <div>Email: {user.email}</div>
      <div>LinkedIn: {user.linkedin_url}</div>
      <div>Slack: {user.slack_id}</div>
    </div>
  );
}
