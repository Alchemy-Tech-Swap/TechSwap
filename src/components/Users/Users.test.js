import Users from './Users';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

test('Should render users profile page', async () => {
  const { container } = render(
    <MemoryRouter>
      <Users
        user={{
          id: 26,
          created_at: '2022-01-20T23:03:53.531704+00:00',
          email: 'rewq@ewq.vfdsa',
          name: 'rewqrewq',
          slack_id: 'rewqrewq',
          linkedin_url: 'rewqrewq',
          user_id: '0fcdf7e2-48e8-4f4d-b4e6-5453ce7ee18d',
        }}
      />
    </MemoryRouter>
  );

  await screen.findByText('Profile', 'Slack: ');

  expect(container).toMatchSnapshot();
});
