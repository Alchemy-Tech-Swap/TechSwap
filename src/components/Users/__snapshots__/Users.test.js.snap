// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render users profile page 1`] = `
<div>
  <div>
    <h1>
      Profile
    </h1>
    <div>
      Name: 
      rewqrewq
    </div>
    <div>
      Email: 
      rewq@ewq.vfdsa
    </div>
    <div>
      LinkedIn: 
      rewqrewq
    </div>
    <div>
      Slack: 
      rewqrewq
    </div>
  </div>
</div>
`;
